src/commonMain/kotlin/operations/numeric/DoubleTypeSensitiveOperation.kt

```diff
@@ -0,0 +1,13 @@
+package operations.numeric
+
+import asDoubleList
+
+internal interface DoubleTypeSensitiveOperation {
+    fun resultOrNull(expression: Any?, operation: (List<Double>) -> Double?): Double? {
```

Comment:
could we find a better name for this function? Currently, we have DoubleTypeSensitiveOperation, if there will be another it will be hard to find what is up. Event doubleResultOrNull will help for sure.